### DISCUSSION TITLE
v0.1.9.2-Beta

### DISCUSSION THREAD

**Discussion by AMacro** [AMacro/dv-multiplayer#57] [Announcements]
# Changes / Fixes

This release is fixes some of the more common issues, it is likely that there are still a number of game-breaking bugs not yet discovered.
Please report any bugs by either raising an issue or visiting the [discord discussion thread](https://discord.com/channels/332511223536943105/1234574186161377363). Be sure to include both the host and client `player.log` files in your report.

## UI
- Fixed minor issues with pinging LAN games that were set to private.
- Added further localisation for previously untranslated English.

## Game
- Fixed Garage unlocking - keys now unlock garages for clients
- Fixed issues with the CareerManager and CommsRadio when the client connects with a new user.
This is also a potential fix for issues with License sync.
- Potential fix for LanternPatch intermittent issue which caused other aspects of the the game to break
- Additional logging to try to trackdown issues with `TrainsOptimizer()` - this issue is intermittent
- Fix for `LocalFruits` error that prevented B98 saves from loading (stuck at 86%) if any loaded cars has LocalFruits cargo

# Known Bugs
- The new tow-a-loosely-hitched-carriage feature is causing some issues, we strongly recommend only towing with the hitch tightened, otherwise a de-sync will occur.
- Clients are unable to self service locomotives - self service objects (e.g. pipes, chutes, plugs, nozzles) not synced
- Horns, whistles and steam loco reversers can play up when more than one player is in the loco; this is not a breaking issue, only a mild inconvenience.
- MU chains behave the same as the horns and whistles, playable but mildly annoying
- Shop, world items, inventory for players other than the host not yet synced
- Many other minor bugs and inconveniences, which we'll work through as we tick off the bigger issues.
- New game features including paint themes and "Demonstrator vehicles" are not completely sync'd and will be worked on over the coming releases.
- Occasionally after leaving a multiplayer session as a client the Continue Session button is missing - this may be fixed but is an intermittent issue and not confirmed.
- When a job is completed and the Job Booklet is turned in by a client, cash only spawns on the host's game - this is expected to be fixed with Item Sync.

# Features in progress
- Job sync - This is on hold, pending release of item sync
- Item sync - good progress has been made; there were a few outstanding issues we were working on prior to B99's release. Major updates in B99 (many new "smart" items and a new mounting system) will delay the release of this feature
- UPnP
- NAT punching

<hr /><em>This discussion was created from the release <a href='https://github.com/AMacro/dv-multiplayer/releases/tag/v0.1.9.2-Beta'>v0.1.9.2-Beta</a>.</em>

**Comment by gercj187** [comment]
Please make it possible to choose between: all players play with same wallet and licenses 
Or each Player have his own wallets and licenses. 

Please let clients also use service terminals, like load/unload cargo and hand over completed jobs.

**Reply by AMacro** [reply]
Hi @gercj187,

At the moment we are focusing on improving stability and fixing some of the issues people have been having since the release of B99.

Service terminals/self-service and cargo load/unload are planned to be added in the near future. Separate wallets and licenses may come further down the track if we can get all of the main features working. We would eventually like to add competitive modes such as 'company mode' which would require multiple wallets or even per-player wallets

**Comment by gercj187** [comment]
Thank you really much for the reply. 
I played now ~ 26 hours in the last two days, and only 4 in the original game.
The last 22 hours are with multiplayer.

This is the most best mod so far and I can't wait to get new stuff...

I have so many ideas and would like mod DV too.
But right now iam struggle with the modding tools.

Modding in Trainz and DayZ was way more easy to learn.

**Reply by AMacro** [reply]
Sorry, just saw you message.

I'm glad you're enjoying it so much!

I haven't had experience with Trainz or DayZ, I had a quick look and it appears Trainz is made with modding in mind, where as DV isn't and requires a bit of C# knowledge as well as some reverse engineering skills

**Comment by gercj187** [comment]
Short question.
How much tris have the Unity-Anime Girl?
Would it be possible to implement a more fitting character?
Someone with orange warn Jacket and helmet?

I found a very nice pack with different jacket, faces, headgear.
I would donate this packs.

**Reply by AMacro** [reply]
To be honest, I’m not sure about the exact tris count; that was part of the mod before I got involved, and I haven’t had much to do with it directly. You may be able to get more info from the [Unity-Chan! asset page](https://assetstore.unity.com/packages/3d/characters/unity-chan-model-18705).

As for implementing a new character, it’s definitely possible. Adding new characters isn’t a priority at the moment since we’re focusing on stabilising the mod and adding core features, however, we currently have someone in the community working on a model and skins; once the character is ready, we’ll do our best to integrate it as soon as we can. 

There will be a learning curve for me with the implementation—I don't have a game dev background and have mostly focused on the coding. While I do appreciate the offer, I wouldn’t want you to spend any money on something I might not be able to get working properly just yet.
